test/e2e/clusterctl_upgrade.go

```diff
@@ -133,10 +134,17 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		} else {
 			clusterctlBinaryURLTemplate = input.InitWithBinary
 		}
+
+		// create InitWithKubernetesVersion field to set the init kubernetes version via struct
+		if input.InitWithKubernetesVersion == "" {
+			Expect(input.E2EConfig.Variables).To(HaveKey(initWithKubernetesVersion), "Invalid argument. %s variable must be defined when calling %s spec", initWithKubernetesVersion, specName)
+			Expect(input.E2EConfig.Variables[initWithKubernetesVersion]).ToNot(BeEmpty(), "Invalid argument. %s variable can't be empty when calling %s spec", initWithKubernetesVersion, specName)
+			input.InitWithKubernetesVersion = input.E2EConfig.GetVariable(initWithKubernetesVersion)
+		} else {
+			input.InitWithKubernetesVersion = os.Getenv(initWithKubernetesVersion)
+		}
 		clusterctlBinaryURLReplacer := strings.NewReplacer("{OS}", runtime.GOOS, "{ARCH}", runtime.GOARCH)
 		clusterctlBinaryURL = clusterctlBinaryURLReplacer.Replace(clusterctlBinaryURLTemplate)
-		Expect(input.E2EConfig.Variables).To(HaveKey(initWithKubernetesVersion))
-		Expect(input.E2EConfig.Variables).To(HaveKey(KubernetesVersion))
```

Comment:
Why was this removed?

Reply:
The checks are already parsed under the if-else statement, is it necessary to keep those too?